packages/venia-ui/lib/components/MagentoRoute/magentoRoute.js

```diff
@@ -14,9 +14,19 @@ const MESSAGES = new Map()
 
 const MagentoRoute = () => {
     const talonProps = useMagentoRoute();
-    const { component: RootComponent, id, isLoading, routeError } = talonProps;
+    const {
+        component: RootComponent,
+        id,
+        isLoading,
+        routeError,
+        redirectCode,
+        relativeUrl
+    } = talonProps;
 
-    if (isLoading) {
+    if (redirectCode === 301 || redirectCode === 302) {
```

Comment:
According to the [urlResolver query docs](https://devdocs.magento.com/guides/v2.4/graphql/queries/url-resolver.html?itm_source=devdocs&itm_medium=search_page&itm_campaign=federated_search&itm_term=urlresolver#output-attributes), this value will only ever be `0`, `301`, or `302` 👍